extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala

```diff
@@ -25,37 +26,32 @@ object GenericBehaviors {
   ): Behavior[I] =
     Behaviors
       .setup[Any] { context => // use any to mimic union types
+
         import context._
 
+        setLoggerName(classTag[I].runtimeClass)
+
         def start(): Behavior[Any] = {
 
-          val adapter = context.messageAdapter[Listing](identity)
+          /*
+            We use child actors instead of message adapters to avoid issues with clustered receptionist
```

Comment:
We need a better description of the clustered receptionist problem.

Reply:
I'll come up with something

Reply:
I left a comment on why messageAdapters can't be use, but this is not specific to this fix but to working with receptionist. I guess there is no point of explaining reason there in depth as this solution is still valid use of actors.